### PR TITLE
:memo: Add commit message guide

### LINKED ISSUE #10: Add description about commit messages
Commit messages deliver a lot of information to a developer. There should be a paragraph dedicated to commit messages in workflows section.

### PR DESCRIPTION
**Pull request by yani-**
Providing guidelines of what a good commit message should look like.

Fixes #10

### CODE REVIEW

**Comment by yani-** [comment]
Live preview: http://yani-.github.io/engineering-best-practices/workflows/#commit-messages

File: _includes/markdown/Workflows.md [hunk 1]
```diff
@@ -79,6 +79,28 @@ Do not wait more than a day to push commits. _Push early and push often._ Not on
 
 Try to keep your commits as logically self-contained as possible. If you have made a lot of changes which you have not committed yet, you can step through the changes via `git add -p` to selectively stage chunks for committing. In this way you can group your changes into commits. Using `git add -p` is also just plain handy to stage changes for commit in general because it helps prevent accidentally committing something you do not intend; for the same reason, `git commit -a` should be avoided because you don't have the explicit step to review what you are committing. Remember you can do `git diff --staged` right before commit for one last review.
 
+##### Commit Messages
+
+The gist of a great commit message is this:
+
+Capitalized, short (50 chars or less) summary
```
Comment (westonruter) [inline _includes/markdown/Workflows.md]: End summary with period?

**Comment by westonruter** [comment]
Should we also reference and glean from the new Commit Message in the Core Handbook? https://make.wordpress.org/core/handbook/best-practices/commit-messages/

**Comment by westonruter** [comment]
- [ ] Another enhancement we can add here is a `commit-msg` hook to enforce proper commit messages: https://github.com/xwp/wp-dev-lib/issues/107

**Comment by westonruter** [comment]
Use of Emoji in commit messages is fun :smile: 

We should make sure that the client is OK with it when using it, nevertheless.

**Comment by yani-** [comment]
@westonruter
> Should we also reference and glean from the new Commit Message in the Core Handbook?

We could post a link to the Core Handbook for a broader description. There is a section talking about props and encouraging giving props:
```
Props should be given to all those who contributed to the final commit, whether through patches, refreshed patches, code suggested otherwise, design, writing, user testing, or other significant investments of time and effort. Usernames are parsed for the credits list and WordPress.org profiles.
```
I think this is only applicable to open source projects. What's your opinion?

File: _includes/markdown/Workflows.md (hunk 1 above)
Comment (yani-) [inline _includes/markdown/Workflows.md]: It's not popular to end summary with a period. Do we want to emphasize that it should be a proper sentence?

Comment (westonruter) [inline _includes/markdown/Workflows.md]: I don't really care, either way. Let's just leave it `undefined` :smile:

**Comment by westonruter** [comment]
> I think this is only applicable to open source projects. What's your opinion?

Agreed. Since everyone working on a project can commit directly to it via Git, there's much less need for props. Sometimes, though, I will explicitly commit something with the `--author` flag to attribute the code to someone else. So this is what should be used in that case, since we can take advantage of Git.

**Comment by yani-** [comment]
@westonruter when you get a moment, I think this PR is ready to go :satisfied:

**Pull request merged**